.github/workflows/pr_verification.yml

```diff
@@ -0,0 +1,52 @@
+---
+name: PR verification workflow
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  checkstyle:
```

Comment:
Looking at the two options and giving it a couple days to stew on it, I think I'm ultimately in favor of this method here where we build everything on a given invocation of the job.

The extra time isn't significant for us, as our other tests are going to dominate the overall runtime by an order of magnitude or two, and addressing problems on a per-run basis seems like it'll be easier than having to go through the mental jogging of remember where the image is and rebuilding it in cases where that is the resolution.

Ideally we can arrive at a place in the future where we're no longer tied to dependencies that have to be installed at the system level to run our various tasks, eliminating the need to make considerations and trade-offs like this.

Reply:
I agree. Extra time isn't that bad and it will save us a lot of maintenance overhead.

Reply:
I'm also in agreement here. The maintenance overhead for the runner image doesn't seem worth the benefit of saving a small amount of time! 

If I understand the cache action properly, we could also cache these dependencies so that they are only downloaded once and then pulled from the cache for each job [0].  However, this adds more complexity to our jobs for not much gain.

[0] https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

Reply:
@JoshAlbrecht23 Caching should be handled automatically by the gradle action. https://github.com/gradle/gradle-build-action

Reply:
Perfect!! Thanks for linking that.